Sources/TuistSupport/Utils/Environment.swift

```diff
@@ -173,4 +197,26 @@ public class Environment: Environmenting {
     public var settingsPath: AbsolutePath {
         directory.appending(component: "settings.json")
     }
+    
+    private func loadEnvironmentVariablesFromFile() throws {
+        guard FileManager.default.fileExists(atPath: ".env") else {
+            throw EnvFileFailure.fileMissing(path: ".env")
+        }
+        guard let contents = try? String(contentsOf: URL(fileURLWithPath: ".env")) else {
+            throw EnvFileFailure.contentsUnreadable
+        }
+        let lines = contents.split(separator: "\n")
+        // skip comment lines
+        for line in lines where !line.hasPrefix("#") {
+            let pair = line.split(separator: "=")
+            guard let key = pair.first?.trimmingCharacters(in: .whitespacesAndNewlines),
+                  let value = pair.last?.trimmingCharacters(in: .whitespacesAndNewlines),
+                  pair.count == 2,
+                  !key.isEmpty,
+                  !value.isEmpty else {
+                throw EnvFileFailure.malformedKeyValuePair
+            }
+            setenv(key, value, 1)
+        }
+    }
```

Comment:
I'd extract this logic into a utility whose only responsibility is loading the file and returning a dictionary with the content:

```swift
protocol EnvironmentFileReading {
  func loadEnvironmentFile(directory: AbsolutePath) throws -> [String: String]
}
class EnvironmentFileReader {

}
```

Then the logic here all it does is using that utility to get the content and set it in the environment using `setenv`.